packages/server/src/AccountsServer.js

```diff
@@ -240,6 +231,32 @@ export class AccountsServer {
   }
 
   /**
+   * @description Server use only. This method creates a session
+   *              without authenticating any user identity.
+   *              Any authentication should happen before calling this function.
+   * @param {UserObjectType} userId - The user object.
+   * @param {string} ip - User's ip.
+   * @param {string} userAgent - User's client agent.
+   * @returns {Promise<LoginReturnType>} - Session tokens and user object.
+   */
+  // eslint-disable-next-line max-len
+  async loginWithUser(user: UserObjectType, ip?: ?string, userAgent?: ?string): Promise<LoginReturnType> {
```

Comment:
These are handled on the using function as they are otherwise hard to control and prevent duplicate events being emitted.